File: BackEnd/EntreBicis/src/main/java/MarioOlaya/Projecte4/EntreBicis/entity/SystemParameter.java
package MarioOlaya.Projecte4.EntreBicis.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Representa els paràmetres del sistema utilitzats per configurar diferents aspectes
 * del funcionament de l'aplicació.
 * 
 * Aquesta entitat inclou informació sobre la velocitat màxima, el temps màxim
 * d'aturada, la conversió de quilòmetres a punts i el temps de caducitat de les recompenses.
 * 
 * @author Mario Olaya
 */
@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "system_parameter")
public class SystemParameter {
    
    @Id
    private Long id;
    
    //velocitat maxima en km/h
    private int maxVelocity;

    //temps maxim de aturada en minuts
    private int maxStopTime;

    //conversio entre km y punts en punts per km
    private int pointsPerKm;

    //tempx de caducitat de la recompensa en hores
    private int rewardExpirationTime;

}

File: FrontEnd/app/src/main/java/cat/copernic/entrebicis/Core/Model/SystemParameter.kt
package cat.copernic.entrebicis.Core.Model

/**
 * Classe de dades que representa els paràmetres globals de configuració del sistema.
 *
 * Aquests paràmetres controlen el comportament de la validació de rutes i el sistema de recompenses.
 *
 * @property id Identificador únic dels paràmetres del sistema.
 * @property maxVelocity Velocitat màxima permesa per validar una ruta (km/h).
 * @property maxStopTime Temps màxim d'aturada (en minuts) abans de considerar una ruta finalitzada automàticament.
 * @property pointsPerKm Nombre de punts obtinguts per cada quilòmetre recorregut.
 * @property rewardExpirationTime Temps de caducitat d'una recompensa (en hores).
 */
data class SystemParameter(
    val id: Long,
    val maxVelocity: Int,
    val maxStopTime: Int,
    val pointsPerKm: Int,
    val rewardExpirationTime: Int
)
